### ISSUE TITLE
P1 - ver84cp1r1 - ae66598d-426b-4c05-bb66-602dc6e39e49

### ISSUE THREAD

**Issue by zbynekwinkler** [robotika/subt-artf#38]
https://subtchallenge.world/submission/ae66598d-426b-4c05-bb66-602dc6e39e49

Test finální strategie - CDE Freyjas, ABFGH drones.

Freyji zatím dojely mega daleko, ale C je zadrhnutá v nějakém kopci kamení, D je na zádech a furt točí kolečkama a jen E stále někam jede (projíždí kolem breadcrumbs, takže už tam někdo byl). Všechny 3 odhazují.

Drona odstartovala zatím jen H a z neznámého důvodu si lítá po startovní oblasti.

Bod máme zatím 1 a zbývá nám 35 pokusů a 37 minut.

**Comment by zwn** [comment]
Drona G odstartovala a snad se jí podařilo alespoň vletět do jeskyně.

**Comment by zwn** [comment]
Freyja C už je také na zádech (jako D). Myslel jsem, že se E osvobodila, ale ona se tam zase vrátila. Vypadá to, že krouží takové relativně malé kolečko - na suť a zpět.

**Comment by zwn** [comment]
I drona A odstartovala (vlevo).

**Comment by zwn** [comment]
Score 2, sim_time 39m56s, 29 reports remaining.
Všechny drony někde lítají, freyjas jsou na tom furt stejně.

**Comment by jisa** [comment]
Ve chvili, kdy odjede Freyja pred dronou H, drone H se rozkmita lokalizace a nakonec se ztrati jeste nez se zacne hybat. To mi prijde jako dost zasadni problem.

Jedno reseni muze byt spravovat lokalizaci, cemuz moc neverim. Druhe muze byt resetovat offset dokud to jde. @zwn @m3d 

![image](https://user-images.githubusercontent.com/686527/95922737-3b97a380-0db4-11eb-880a-bf09b28c7cf0.png)

**Comment by zwn** [comment]
Ten log je mega-velký (3GB). Pokud budeme na začátku nějak výrazněji čekat, tak se taky nabízí otázka, jestli to čekání nemá být někde vně toho vlastního řešení - tj. někde, kde to vlastně ještě vůbec neběží, nevzniká žádný log a taky není šance na zmatení.

**Comment by zwn** [comment]
Periodické korekce podle offsetu to asi udrží:

![image](https://user-images.githubusercontent.com/1130051/95959983-796fe880-0e03-11eb-8b9c-81f22098519b.png)

Takhle to vypadá, když místo `fromrospy.pose3d` nakreslím `offseter.pose3d`, který koriguje průběžně.

**Comment by zwn** [comment]
Jo, je to situace těsně před tím než se vznese, tj. reálný čas 8h33m35s.

**Comment by jisa** [comment]
Mam sklony smazat vsechny vyskyty slova "offset" v main.py a posilat tam offseter.pose3d. Namitky?

Pokud to nekdo udela uz behem dne, tim lip. Akorat bych rad varovani predem, ze se to deje, at to nedelame dvakrat.

Druhotne premyslim o jednom z predchozich Zbynkovych navrhu: Pridat tam dalsi ROS node, ktery se z pohledu Kalmana tvari ve startovni oblasti jako GPS. Offsetter totiz jakz tazk vyresi xyz, ale odhady rychlosti, smeru a rotacnich rychlosti muzou zustat rozbite.

Mimochodem, to rozbiti lokalizace asi nejak souvisi s pohybem te odjizdejici Freyji. Do te doby to neni tak zle. Pri lokalnim behu, kdy necham dronu dlouho na startu a nic jineho tam neni, oddriftovala o metr a pul za pul simulovane hodiny, coz je mnohem mene.

**Comment by zwn** [comment]
> Mam sklony smazat vsechny vyskyty slova "offset" v main.py a posilat tam offseter.pose3d. Namitky?

:+1: Já už to zkoušel tímto směrem posouvat, ale nezdařilo se. Tj. já ti rozhodně bránit nebudu.

> Druhotne premyslim o jednom z predchozich Zbynkovych navrhu: Pridat tam dalsi ROS node, ktery se z pohledu Kalmana tvari ve startovni oblasti jako GPS. Offsetter totiz jakz tazk vyresi xyz, ale odhady rychlosti, smeru a rotacnich rychlosti muzou zustat rozbite.

Ano, proto mi to přišlo jako nejlepší řešení, na které jsem se ale implementačně necítil - proto alespoň offseter.

Další možné řešení je čekat vně pythonu a ne uvnitř - tj. po spuštění pythonu se prostě vyrazí bez dalšího čekání (převedení na předchozí případ). Ale je to takový hack.

**Comment by m3d** [comment]
p.s. skoro si myslim, ze v prvnim kole (uz za to nelobuji) muze ten `offset` v `subt/main.py` zustat a "jenom" pripravit tu GPS integraci - mel by se pak detekovat jako nulovy a nemel by napachat skody ... snad. Ale @zwn by zase byl happy, kdyby zmizel ... jen nevim zda dnes.

**Comment by zwn** [comment]
> Ale @zwn by zase byl happy, kdyby zmizel ... jen nevim zda dnes.

Na mě se neohlížejte, já řeším jiné věci - avizoval jsem, že v tomto směru se nebudu nijak angažovat.

**Comment by tajgr** [comment]
![state tlog](https://user-images.githubusercontent.com/6401714/95997821-7c370180-0e34-11eb-81f0-87352c53fe8d.png)

**Comment by tajgr** [comment]
Artefakty jsme viděli a detekovali: lano, batoh, telefon a helma (Freyja C, pravděpodobně se report nedostal z jeskině).
Falešné detekce: 5 x helma, 1x telefon, 1 x batoh. Žádná detekce ze startovní oblasti!

**Comment by tajgr** [comment]
![im_08171](https://user-images.githubusercontent.com/6401714/96024750-751feb80-0e54-11eb-88fc-6f52f0d684f8.jpg)

Tohle lano (schované za tečkami) detekujeme cca na 30 po sobě jdoucích obrázcích. Přes validaci skrze depth data se ale dostanou jen tři poslední obrázky (vzdálenost je cca 5 m). Spoluviník se zdá být příliš nízká hodnota thresholdu pro lano. Motivace pro nízkou hodnotu byla citlivá detekce, ale pokud to způsobuje toto, tak je efekt spíš opačný.

**Comment by tajgr** [comment]
![im_08171](https://user-images.githubusercontent.com/6401714/96035787-d13e3c00-0e63-11eb-891d-3ec1d81f9df2.jpg)

Tak asi beru zpět. Nezdá se, že by to mělo až tak velký význam. Zelené tečky představují body se score > 0.7. Musel bych to posunout až na 0.9, aby to mělo nějaký větší význam.
